pkg/operator/ceph/object/admin.go

```diff
@@ -110,8 +111,57 @@ func runAdminCommand(c *Context, expectJSON bool, args ...string) (string, error
 			fmt.Sprintf("--rgw-zone=%s", c.Zone),
 		}
 
-		return RunAdminCommandNoMultisite(c, expectJSON, append(args, options...)...)
+		args = append(args, options...)
 	}
 
-	return RunAdminCommandNoMultisite(c, expectJSON, args...)
+	// work around https://github.com/rook/rook/issues/7573 (FIFO file I/O issue)
+	result, err := RunAdminCommandNoMultisite(c, expectJSON, args...)
+	if isFifoFileIOError(err) {
+		command := ""
+		if len(args) >= 2 {
+			command = fmt.Sprintf(" %s %s", args[0], args[1]) // e.g., "user create" or "user update"
+		}
+		logger.Infof("retrying 'radosgw-adimin%s' command with OMAP backend to work around FIFO file I/O issue", command)
+
+		// We can either run 'ceph --version' and then pick a flag to use, or we can just try to
+		// use both flags and return the one that works. Same number of commands being run.
+		retryArgs := append(args, "--rgw-data-log-backing=omap") // v16.2.0-
+		retryResult, retryErr := RunAdminCommandNoMultisite(c, expectJSON, retryArgs...)
+		if isInvalidFlagError(retryErr) {
+			retryArgs = append(args, "--rgw-default-data-log-backing=omap") // v16.2.1+
+			retryResult, retryErr = RunAdminCommandNoMultisite(c, expectJSON, retryArgs...)
+		}
+
+		return retryResult, retryErr
+	}
+
+	return result, err
+}
+
+func isFifoFileIOError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	exitCode, extractErr := exec.ExtractExitCode(err)
+	if extractErr != nil {
+		logger.Infof("failed to determine return code of 'radosgw-admin' command. assuming this could be a FIFO file I/O issue. %#v", err)
```

Comment:
```suggestion
		logger.Errorf("failed to determine return code of 'radosgw-admin' command. assuming this could be a FIFO file I/O issue. %#v", err)
```

?